src/modules/core/components/avatar/avatar.tsx

```diff
@@ -0,0 +1,82 @@
+import { type FC, useMemo } from 'react';
+import clsx from 'clsx';
+import Image from 'next/image';
+
+import type { AvatarProps } from './avatar.interface';
+import styles from './avatar.module.scss';
+
+export const Avatar: FC<AvatarProps> = ({
+  url,
+  size = 'small',
+  fallback = <UserIcon />,
+  shadow = false,
+  shape = 'circle',
+}) => {
+  const avatarSize = useMemo(() => {
+    const SIZES = {
+      medium: 56,
+      large: 156,
+      small: 40,
+    };
+
+    if (size === 'medium') {
+      return SIZES.medium;
+    } else if (size === 'large') {
+      return SIZES.large;
+    }
+
+    return SIZES.small;
+  }, [size]);
+
+  return (
+    <div
+      className={clsx(styles.avatarContainer, styles[shape], {
+        [styles.shadow]: shadow,
+      })}
+      style={{
+        width: `${avatarSize}px`,
+        height: `${avatarSize}px`,
+      }}
+    >
+      {url ? (
```

Comment:
I suggest using different variants `{url && <Image .... />}` and `{!url && fallback && <div ... />}`.